### DISCUSSION TITLE
Help with Critical WordPress Error after update

### DISCUSSION THREAD

**Discussion by roysinn** [liedekef/events-made-easy#615] [General]
My website is now up but with EME deactivated. Today's update to Version 2.5.3 caused website to generate with "There has been a critical error on this website.". Information provided by WordPress was: An error of type E_PARSE was caused in line 115 of the file /home/coscowor/public_html/wp-content/plugins/events-made-easy/eme-fs.php. Error message: syntax error, unexpected token "{"

I deactivated EME to keep my website up and running. I can survive until August 9. Just need to know how to recover and get back into EME. Don't want to loose events I've set up. 

FYI - I've kept Version 2.3.63. Didn't know if I could or should kill it. May have nothing to do with this my latest problem but, if not, can I kill it without corrupting the 2.5.4? 

Cheers,
Roy

**Comment by DanielHoviss** [comment]
Same here. Latest version causes fatal error. Also had version 2.3.63 not active. Would like to clean up and resolve this error.
https://blacksheepradio.org/

When seeking help with this issue, you may be asked for some of the following information:
WordPress version 6.6.1
Active theme: Weaver Xtreme (version 6.5.1)
Current plugin: Events Made Easy (version 2.5.3)
PHP version 8.2.21



Error Details
=============
An error of type E_PARSE was caused in line 115 of the file /home/blacksh4/public_html/wp-content/plugins/events-made-easy-main/eme-fs.php. Error message: syntax error, unexpected token "{"

**Comment by DanielHoviss** [comment]
Here is the corrected php code - replace line 111 to 126...
}
            // location also needs id, latitude and longitude (these are hidden anyway)
			if ( strstr( $field, 'location_' ) ) {
				if (!$location_id_added) {
					$replacement .= eme_get_fs_field_html("location_id");
					$location_id_added = 1;
				}
				if (!$latitude_added) {
					$replacement .= eme_get_fs_field_html("location_latitude");
					$latitude_added = 1;
				}
				if (!$longitude_added) {
					$replacement .= eme_get_fs_field_html("location_longitude");
					$longitude_added = 1;
				}
			}

**Comment by liedekef** [comment]
Sorry, new version released that fixes the typo.

**Comment by CarlAnto** [comment]
Updating the plugin resolved the issue for me!

**Comment by DanielHoviss** [comment]
Great. Perhaps the developers caught the typo.

**Comment by DanielHoviss** [comment]
Great thanks.

**Comment by roysinn** [comment]
Many thanks Daniel, Franky and Carl. 

I'm still stuck. I deactivated 2.5.3 and could not reactivate using Daniel's patch - Fatal Error on activation. I tried to install 2.5.4 from zip but server is timing out (not due to EME but an issue with my server, I believe). I tried to replace just the eme-fs from 2.5.4 but still get a fatal error activating it. 

Question: With EME 2.5.3 deactivated (and assuming my server is fit), should I be able to replace it with the 2.5.4 zip without losing everything? 

Any suggestions would be muchly appreciated.

**Reply by CarlAnto** [reply]
I deactivated all plugins from the php database and updated EME with the zipfile liedekef provided before reinstalling/activating all the plugins.

I currently use php version 8.3 with WordPress 6.6.1

PS: When u deactivate the plugins, your website will be usable but functionality is lost for the pages that use one of your plugins.

**Reply by roysinn** [reply]
Oh. NEVER thought to deactivating all all plug-ins. Thanks for the suggestion and information. I will recheck my php version. I think I'm on 8 point something but I'll recheck.

**Reply by roysinn** [reply]
We're on WordPress 6.6.1 and PHP 8.1.29. Not sure if PHP 8.3 will fix this but great to know that I can safely upgrade. Thanks Carl.

**Comment by liedekef** [comment]
Weird that the eme-fs.php file from 2.5.4 would give issues, that one should be (and is) correct. Are you sure you actually replaced the file? No caching plugin that interferes?
Overwriting the 2.5.3 with the 2.5.4 version will work without problems. But try again with just the file from 2.5.4, it really is ok (I upgraded several sites to 2.5.4 without problems).

**Reply by roysinn** [reply]
Thanks Franky. I'll recheck that I have the new file and clear the cache. Good to know that that I can replace with zip. Server is deadly slow right now so I'll explore that for a Plan "B".

**Reply by liedekef** [reply]
if you have issues, let me know and give me temp admin access, I'll then fix the file for you.

**Reply by roysinn** [reply]
Thank you Franky and thank you, everyone, for your help. Things are looking up. I cloned my WP site, reinstalled the latest version of EME and restored the tables from a backup. everything seems to work except the EME global Settings was not restored. All the EME tables are prefixed with "wpfx_eme_" and I thought I copied them all. Maybe because the cloned site URL is different? Any suggestions? 

Not a huge deal if I have to type in the settings but it would help if there was some way to restore the settings from my backup. 

Cheers.

**Reply by liedekef** [reply]
The global EME settings are stored in the generic WP options table. So if you cloned that one, EME should just take over those settings. All the EME settings in that table start with option name "eme_", so you could just export those rows , remove the existing eme_* rows from your new install and then import the exported ones. Or just set the settings again :-)

**Reply by roysinn** [reply]
Ah! I didn't copy the settings from the generic WP options table. Things are looking up! 
Thank you so much for the prompt answer. Learning a lot for this unexpected lesson. 

Cheers,
Roy

**Comment by roysinn** [comment]
Franky – that is so generous of you. I’ll try my best to fix the problem this evening and let you know. 

 

Cheers,

Roy
